android/src/main/java/com/bitmovin/player/reactnative/AnalyticsModule.kt

```diff
@@ -11,172 +10,35 @@ private const val MODULE_NAME = "AnalyticsModule"
-    fun setCustomData(nativeId: NativeId, playerId: NativeId?, json: ReadableMap?) {
-        uiManager()?.addUIBlock { _ ->
-            val source = playerModule()?.getPlayer(playerId)?.source
-            val collector = collectors[nativeId]
-            val customData = JsonConverter.toAnalyticsCustomData(json)
-            when {
-                source == null -> Log.d(
-                    "[AnalyticsModule]", "Could not find source for player ($playerId)"
-                )
-                collector == null -> Log.d(
-                    "[AnalyticsModule]", "Could not find analytics collector ($nativeId)"
-                )
-                customData == null -> Log.d(
-                    "[AnalyticsModule]", "Could not convert custom data, thus they are not applied to the active source for the player ($playerId) with the collector ($nativeId)"
-                )
-                else -> collector.setCustomData(source,  customData)
-            }
-        }
-    }
-
-    /**
-     * Gets the current custom data config for a `BitmovinPlayerCollector` instance.
-     * @param nativeId Native Id of the the collector instance.
-     * @param promise JS promise object.
-     */
-    @ReactMethod
-    fun getCustomData(nativeId: NativeId, playerId: NativeId?, promise: Promise) {
-        uiManager()?.addUIBlock { _ ->
-            val source = playerModule()?.getPlayer(playerId)?.source
-            val collector = collectors[nativeId]
-            when {
-                source == null -> promise.reject(
-                    "[AnalyticsModule]", "Could not find source for player ($playerId)"
-                )
-                collector == null -> promise.reject(
-                    "[AnalyticsModule]", "Could not find analytics collector ($nativeId)"
-                )
-                else -> promise.resolve(JsonConverter.fromAnalyticsCustomData(collector.getCustomData(source)))
-            }
-        }
-    }
-
-    @ReactMethod
-    fun addSourceMetadata(nativeId: NativeId, playerId: NativeId?, json: ReadableMap?) {
-        uiManager()?.addUIBlock { _ ->
-            val source = playerModule()?.getPlayer(playerId)?.source
-            val collector = collectors[nativeId]
-            val sourceMetadata = JsonConverter.toAnalyticsSourceMetadata(json)
-            when {
-                source == null -> Log.d(
-                    "[AnalyticsModule]", "Could not find source for player ($playerId)"
-                )
-                collector == null -> Log.d(
-                    "[AnalyticsModule]", "Could not find analytics collector ($nativeId)"
-                )
-                sourceMetadata == null -> Log.d(
-                    "[AnalyticsModule]", "Could not convert source metadata, thus they are not applied to the collector ($nativeId)"
-                )
-                else -> collector.addSourceMetadata(source,  sourceMetadata)
+                playerModule()?.getPlayer(nativeId)?.analytics?.sendCustomDataEvent(it)
```

Comment:
Should this be implemented on `PlayerModule` now?
Do we still need `AnalyticsModule`?

Reply:
In favour of not making the `PlayerModule` and `SourceModule` unnecessarily big I do like the separation into a separate `...Module`.

What do you think about renaming the `AnalyticsModule` to `PlayerAnalyticsModule` and introducing a `SourceAnalyticsModule` once we expose the API to react?

Reply:
That is a good point.
I really like the idea of `PlayerAnalyticsModule`!